datasets/pl/wanted/crawler.py

```diff
@@ -0,0 +1,160 @@
+        "date_of_birth": "//p[contains(text(), 'Data urodzenia:')]/strong/text()",
+        "alias": "//p[contains(text(), 'Pseudonim:')]/strong/text()",
+        "citizenship": "//p[contains(text(), 'Obywatelstwo:')]/strong/text()",
+        "height": "//p[contains(text(), 'Wzrost:')]/strong/text()",
+        "eye_color": "//p[contains(text(), 'Kolor oczu:')]/strong/text()",
+    }
+    info = dict()
+    for key, xpath in details.items():
+        q = doc.xpath(xpath)
+        if q:
+            text = q[0].strip()
+            if text != "-":
+                info[key] = text
+
+    person.add("sourceUrl", url)
+    person.add("topics", "crime")
+    person.add("topics", "wanted")
+    # TODO(Leon Handreke): What does country mean in that context? The za_wanted crawler also does this.
+    person.add("country", "pl")
+
+    h.apply_name(person, full=info["full_name"], middle_name=info.get("middle_name"))
+    h.apply_date(person, "birthDate", info.get("date_of_birth"))
+    person.add("birthPlace", info.get("place_of_birth"))
+    person.add("gender", info.get("gender"))
+    person.add("alias", info.get("alias"))
+    person.add("fatherName", info.get("father_name"))
+    person.add("motherName", info.get("mother_name"))
+    # TODO(Leon Handreke): Add mother maiden name or is that too detailed?
+    person.add("motherName", info.get("mother_maiden_name"))
+    person.add("height", info.get("height"))
+    person.add("eyeColor", info.get("eye_color"))
+
+    citizenship = info.get("citizenship")
+    cleaned_citizenship = (
+        clean_and_split_citizenship(citizenship) if citizenship else None
+    )
+
+    person.add("citizenship", cleaned_citizenship, original_value=citizenship)
+
+    crimes = doc.xpath(
+        "//p[contains(text(), 'Podstawy poszukiwań:')]/following-sibling::ul//a/text()"
+    )
+    if not crimes:
+        context.log.warn("No crimes found for person", entity_id=person.id, url=url)
+    # TODO(Leon Handreke): Every wanted list puts this in notes -- could there be a better place?
+    person.add("notes", crimes)
+
+    context.emit(person, target=True)
+
+
+def crawl_index(context, url) -> str | None:
+    context.log.info(f"Crawling index page {url}")
+    doc = context.fetch_html(url, cache_days=1)
+    # makes it easier to extract dedicated details page
+    doc.make_links_absolute(context.dataset.data.url)
+    cells = doc.xpath("//li[.//a[contains(@href, '/pos/form/r')]]/a/@href")
+    for cell in cells:
+        crawl_person(context, cell)
+
+    # On the last page, the next button will not have an <a>, so this will not match
+    next_button_href = doc.xpath(
+        "//li/a/span[contains(text(), 'następna')]/parent::a/@href"
+    )
+    return next_button_href[0] if next_button_href else None
+
+
+def crawl(context):
+    next_url = context.dataset.data.url
+    # Use this construction instead of recursion because Python sets a recursion limit
```

Comment:
if python had tail recursion 🥹

Reply:
First thing I searched for, apparently Guido has opinions on this that I did not care to investigate further...